### ISSUE TITLE
Massive performance regression on `update` in 4.17

### ISSUE THREAD

**Issue by Inok** [liquibase/liquibase#3675]
## Environment

**Liquibase Version**: 4.17+ (OpenJDK 17)

**Liquibase Integration & Version**: CLI

**Liquibase Extension(s) & Version**: 

**Database Vendor & Version**: PostgreSQL 14.5

**Operating System Type & Version**: Alpine 3.16

**Infrastructure Type/Provider**: -

## Description

Liquibase 4.17 has massive performance regression on `update`. Look at the timings on different versions:

4.16.1 takes only 15s to run:
```
Starting Liquibase at 16:18:39 (version 4.16.1 #4594 built at 2022-09-14 15:27+0000)
Liquibase Version: 4.16.1
Liquibase Community 4.16.1 by Liquibase
[2023-01-11 16:18:39] INFO [liquibase.database] Set default schema name to public
[2023-01-11 16:18:39] INFO [liquibase.lockservice] Successfully acquired change log lock
[2023-01-11 16:18:45] INFO [liquibase.changelog] Reading from public.databasechangelog
[2023-01-11 16:18:52] INFO [liquibase.lockservice] Successfully released change log lock
Liquibase command 'update' was executed successfully.
```

4.17.0 takes **almost 2 minutes** (1m 46s) for the same run:
```
Starting Liquibase at 16:23:21 (version 4.17.0 #4922 built at 2022-10-05 14:56+0000)
Liquibase Version: 4.17.0
Liquibase Community 4.17.0 by Liquibase
[2023-01-11 16:23:21] INFO [liquibase.database] Set default schema name to public
[2023-01-11 16:23:21] INFO [liquibase.lockservice] Successfully acquired change log lock
[2023-01-11 16:23:29] INFO [liquibase.changelog] Reading from public.databasechangelog
[2023-01-11 16:25:07] INFO [liquibase.lockservice] Successfully released change log lock
Liquibase command 'update' was executed successfully.
```

The latest version (4.18) is also affected.

## Traces
I collected traces (`--monitor-performance=true`) from both versions: [perf-jfr.zip](https://github.com/liquibase/liquibase/files/10394237/perf-jfr.zip)
Hope it'll help to resolve the issue.

**Comment by kevin-atx** [comment]
Hey @filipelautert  - since this is a performance issue and Inok has included a performance trace file, could you review/triage this ticket and let us know what you find out?

**Comment by filipelautert** [comment]
Hello @Inok - thanks for the report!
We reviewed the traces and I think we may find the point were there the problem is. I implemented a simple test fix on PR #3684 . Here are the artifacts for it: https://github.com/liquibase/liquibase/suites/10339006499/artifacts/509167433

Could you please use those artifacts and check if the problem is resolved ?

Thanks
